src/library/scala/collection/StringParsers.scala

```diff
@@ -0,0 +1,283 @@
+package scala
+package collection
+
+import scala.annotation.tailrec
+
+/** A module containing the implementations of parsers from strings to numeric types, and boolean
+ */
+final private[scala] object StringParsers {
+
+  //compile-time constant helpers
+
+  //Int.MinValue == -2147483648
+  private final val intOverflowBoundary = -214748364
+  private final val intOverflowDigit = 9
+  //Long.MinValue == -9223372036854775808L
+  private final val longOverflowBoundary = -922337203685477580L
+  private final val longOverflowDigit = 9
+
+  
+  @inline
+  private def decValue(ch: Char): Int = java.lang.Character.digit(ch, 10)
+
+  //bool
+  @inline
+  def parseBool(from: String): Option[Boolean] = from.toLowerCase match {
+    case "true" => Some(true)
+    case "false" => Some(false)
+    case _ => None
+  }
+
+  //integral types
+  @inline
+  def parseByte(from: String): Option[Byte] = {
+    val len = from.length()
+    @tailrec
+    def step(i: Int, agg: Int, isPositive: Boolean): Option[Byte] = {
```

Comment:
This method is basically identical between Byte and Short.  You should make it a private[this] method (maybe even using the Int variant if it isn't significantly slower) and check the range afterwards.

Reply:
@Ichran can you maybe help me with the benchmarks?

From the results I'm getting, `"100".toByte` is *slower* than

```
try {
  Some("100".toByte)
}
catch {
  case e: NumberFormatException => None
}
```

(on my machine   13.244 +/-  0.321 ns/op for the plain case vs 12.697 +/-  0.180 ns/op for the try  catch with 10 warmup iterations and 30 measurement iterations of 1 second each)